core/src/main/java/com/datastrato/gravitino/catalog/BaseCatalog.java

```diff
@@ -27,6 +30,11 @@
  */
 public abstract class BaseCatalog<T extends BaseCatalog>
     implements Catalog, CatalogProvider, HasPropertyMetadata {
+  private static final Logger LOG = LoggerFactory.getLogger(BaseCatalog.class);
+
+  // A hack way to inject custom operation to Gravitino, the object you used is not stable, don't
+  // use it unless you know what you are doing.
+  @VisibleForTesting public static final String CATALOG_OPERATION_IMPL = "ops-impl";
```

Comment:
If this is visible for test, I think package private is enough, no need to be public.

Reply:
this's is public because it's used by `CatalogHiveIT`

Reply:
I think you need to put the property into catalog property metadata to manage the properties altogether.